src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Management
+{
+    public class InstanceManager : IInstanceManager
+    {
+        private static AssignmentContext _assignmentContext;
+        private static object _assignmentLock = new object();
+
+        private readonly ScriptHostManager _scriptHostManager;
+        private readonly WebHostSettings _webHostSettings;
+        private readonly ILogger _logger;
+
+        public InstanceManager(WebScriptHostManager scriptHostManager, WebHostSettings webHostSettings, ILoggerFactory loggerFactory)
+        {
+            _scriptHostManager = scriptHostManager;
+            _webHostSettings = webHostSettings;
+            _logger = loggerFactory.CreateLogger(nameof(InstanceManager));
+        }
+
+        public bool TryAssign(AssignmentContext assignmentContext)
```

Comment:
We need a test for this, verifying app settings applied, and zip unpacked correctly, etc.

Reply:
I added few tests for encrypt/decrypt and TryAssign. 

For testing zip unpacking. Should I use an actual zip? I probably can add a zip to a storage account for this test case. Is that what you were thinking?

Reply:
We can not block this PR on a full test for this, but it will be good to have one. I was thinking a test that points to a temp functions directory, builds a zip as part of test setup, deploys it, etc. A full specialization test that catches regressions and allows us to make changes and step through them easily. Kind of like the StandbyMode_EndToEnd test that I have already.